tool/grind.dart

```diff
@@ -271,3 +272,52 @@ ArchiveFile _fileFromString(String path, String contents,
 ArchiveFile _file(String target, String source, {bool executable: false}) =>
     _fileFromBytes(target, new File(source).readAsBytesSync(),
         executable: executable);
+
+/// Generate a [algo] hash value using the contents of a given [file]
+String _hash(String file, String algo) {
+  Hash hasher;
+  switch (algo) {
+    case 'md5':
+      hasher = md5;
+      break;
+    case 'sha1':
+      hasher = sha1;
+      break;
+    case 'sha256':
+      hasher = sha256;
+      break;
+  }
+  return hasher.convert(new File(file).readAsBytesSync()).toString();
+}
```

Comment:
You're only ever using `sha256` here, so it's probably cleaner to just make this a `_sha256()` function.